src/elements/Table/Table.tsx

```diff
@@ -1,23 +1,37 @@
-import type { TableNode } from '@prezly/story-content-format';
+import type { TableCellNode, TableNode } from '@prezly/story-content-format';
 import classNames from 'classnames';
-import type { HTMLAttributes } from 'react';
+import { Children, cloneElement, type HTMLAttributes, isValidElement } from 'react';
 
-import { TableContextProvider } from './TableContext';
+import { isHeaderCell } from './TableContext';
 
 interface Props extends HTMLAttributes<HTMLTableElement> {
     node: TableNode;
 }
 
 export function Table({ children, node }: Props) {
+    function isHeader(cell: TableCellNode): boolean {
+        return isHeaderCell(node, cell);
+    }
+
     return (
-        <TableContextProvider table={node}>
-            <table
-                className={classNames('prezly-slate-table', {
-                    'prezly-slate-table--withBorders': node.border,
+        <table
+            className={classNames('prezly-slate-table', {
+                'prezly-slate-table--withBorders': node.border,
+            })}
+        >
+            <tbody>
+                {Children.map(children, (child) => {
+                    if (isValidElement(child)) {
+                        return cloneElement<{ isHeader?: (node: TableCellNode) => boolean }>(
```

Comment:
I'm not sure this is the best solution but it works.

I was thinking maybe we could re-use the `Selector` component here, but then you'd have to re-declare all the underlying components here as well, so cloning the children and providing the `isHeader` function felt like the best approach (that I know of) without overspending time.

Reply:
For handling this error, you'd need to type children like so:

```
interface Props extends Omit<HTMLAttributes<HTMLTableElement>, 'children'> {
    children: ReactElement<{ isHeader: any; }>;
    node: TableNode;
}
```
it will most likely result in types erorr in another place.

---

I think `getIsHeader` would be a better name. `isHeader` sounds like a boolean

---

🤔 all this solution sounds quite complex, especially if people would need to create custom components out of it.

what about providing more details in whole document? it can be mapped before rendering, and we can populate table's cells nodes with `isHeader` boolean. I see, we already do some [transformations here](https://github.com/prezly/content-renderer-react-js/blob/feature/care-5792-refactor-table-component-in-content-renderer/src/Renderer.tsx#L54-L57).

Reply:
Yeah, you're right.

I think the transformation is probably the best place to do it - we feed it the table node and then it iterates over the rows and cells and applies the `isHeader` property to them.